axum-extra/src/typed_header.rs

```diff
@@ -1,10 +1,16 @@
-use crate::extract::FromRequestParts;
-use async_trait::async_trait;
-use axum_core::response::{IntoResponse, IntoResponseParts, Response, ResponseParts};
-use headers::HeaderMapExt;
-use http::request::Parts;
+//! Extractor and response for typed headers.
+
+use axum::{
+    async_trait,
+    extract::FromRequestParts,
+    response::{IntoResponse, IntoResponseParts, Response, ResponseParts},
+};
+use headers::Header;
+use http::{request::Parts, HeaderValue};
 use std::{convert::Infallible, ops::Deref};
 
+pub use headers;
```

Comment:
I think it would be more convenient for this to be `axum_extra::headers` instead of `axum_extra::typed_header::headers`. Then the `typed_header` module also doesn't need to be public either, I think.

Reply:
Also should probably be `#[doc(no_inline)]` like before.

Reply:
I'm okay with exporting `headers` and `TypedHeader` from the root but I wanna keep the module public because it contains a bunch more types (like the rejection). I'd rather not expose those from the root.

Time to update all the docs again 😅